### ISSUE TITLE
Feture Request: Unencrypted mode

### ISSUE THREAD

**Issue by Che15ea** [timvisee/ffsend#142]
Hi I am a Tor Browser user and I find it very likely to collapse or fail when encrypting in browser for large files. There is also a size limit that Tor Browser can decrypt(a few hundred MB). I never trust front-end encryption anyway. I always encrypt locally myself. I think it would good to introduce a no-javascript unencrypted mode to serve all user's needs. Thank you!

**Comment by timvisee** [comment]
Send is a primarily a browser based file sharing platform. There's no way to implement an end-to-end encrypted system without served cryptography on the front-end.

`ffsend` (this repository) does not use any JavaScript at all, though. Cryptography is implemented in the tool itself. All encryption/decryption is done on your machine. If you trust its implementation (see source), there's no reason not-to-trust encryption. A Send server has no control over encryption when using `ffsend`.

If you somehow don't trust the cryptography, you can always encrypt the file before uploading through Send or sending through `ffsend`.

I see no reason to implement an unencrypted mode.

**Comment by Che15ea** [comment]
Thank you for your reply! What if the server operator changes the source code? There is no way to verify about that, not like local client which can be verified with PGP.

**Comment by timvisee** [comment]
> Thank you for your reply! What if the server operator changes the source code? There is no way to verify about that, not like local client which can be verified with PGP.

That is true. I've stated this in the bottom of the Security section here: https://github.com/timvisee/ffsend#security

**Comment by Che15ea** [comment]
Since there are some people who don't trust and really rely on front end encryption and decide to encrypt by themselves. I think it is reasonable to implement a unencrypted mode download/upload resumption in order to share big file. There is no other way to share big file stably for tor users.
BTW, it is possible to implement an option not to zip the files, and put them in one link with one password. We may split a big file into dozens and upload and download one by one. Zipping it just made it meaningless.

**Comment by artemislena** [comment]
L: As the admins of an instance, we'd be decidedly against this. Allowing unencrypted uploads that can be downloaded without extra authentication will inevitably result in trolls uploading CSAM or other illegal materials to your server and then reporting the link to it to your registrar or hosting provider, in an attempt to shut down your site and possibly get you involved in lawsuits.

(it's happened a lot on Matrix before Synapse made authenticated fetch the default for media, thankfully we weren't targeted back then)
